src/berry_mill/kiwrap.py

```diff
@@ -1,83 +1,80 @@
+from lxml import etree
+from urllib.parse import ParseResult, urljoin, urlparse
 from typing import Dict
-from berry_mill.kiwiapp import KiwiAppLocal, KiwiAppBox
-from kiwi.kiwi import KiwiError
-from platform import machine
-
-class KiwiParams(TypedDict):
-    """
-    Dictionary for Kiwi Parameters
-
-    Attributes:
 
-        profile (str, Optional): select profile for images that makes use of it.
-        box_memory (str, Default: 8G): specify main memory to use for the QEMU VM (box).
-        debug (bool, Default: False): run in debug mode. This means the box VM stays open and the kiwi log level is set to debug(10).
-        clean (bool, Default: False): cleanup previous build results prior build.
-        cross (bool, Default: False): cross image build on x86_64 to aarch64 target.
-        cpu (str, Optional): cpu to use for the QEMU VM (box)
-        local (bool, Default: False): run build process locally on this machine. Requires sudo setup and installed KIWI toolchain.
-        target_dir (str, Default:/tmp/IMAGE_NAME[.PROFILE_NAME]): store image results in given dirpath.
-    """
-    profile: str
-    box_memory: str
-    debug: bool
-    clean: bool
-    cross: bool
-    cpu: str
-    local: bool
-    target_dir: str
+from berry_mill.params import KiwiParams
 
 
-class KiwiBuilder:
+class KiwiParent:
     """
-    Kiwi wrapper for appliance builder.
+    Base Class for Kiwi Tasks, such as build and prepare
+    Implements shared functionality needed for all children
+    such as repository and global kiwi params handling
     """
-    # TODO handle TypeError
-    def __init__(self, descr:str, **kw: Unpack[KiwiParams]):
+    def __init__(self, descr:str, **pkw: Unpack[KiwiParams]):
         self._repos:Dict[str, Dict[str, str]] = {}
         self._appliance_path:str = os.getcwd()
-        print(self._appliance_path)
         self._appliance_descr:str = descr
-        self._params:Dict[KiwiParams] = kw
-
-        if self._params.get("target_dir"):
-            self._params["target_dir"] = self._params["target_dir"].rstrip("/")
 
         self._trusted_gpg_d:str = "/etc/apt/trusted.gpg.d"
 
         self._tmpdir:str = tempfile.mkdtemp(prefix="berrymill-keys-", dir="/tmp")
 
-        self._boxrootdir:str = os.path.join(self._appliance_path, "boxroot")
-        print(self._boxrootdir)
-        self._fcleanbox:bool = False
-        # tmp boxroot dir only needed when build mode is not local
-        if not self._params.get("local", False):
-            boxdir:str = os.path.join(self._appliance_path, "boxroot")
-            if not os.path.exists(boxdir):
-                # flag for cleanup to remember to also delete boxroot dir if
-                # it hasnt existed before already
-                self._fcleanbox = True
-                os.makedirs(boxdir)
+        self._kiwiparams:Dict[KiwiParams] = pkw
+
+        self._kiwi_options:List[str] = []
+
+        self._initialized:bool = False
+
+        if self._kiwiparams.get("debug", False):
```

Comment:
Replacing all that stuff with `argparse` would be a much better idea.

Reply:
Still dont know what you mean